sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/AnalyticsTransmissionTarget.java

```diff
@@ -66,12 +84,61 @@ public synchronized AnalyticsTransmissionTarget getTransmissionTarget(String tra
         /* Reuse instance if a child with the same token has already been created. */
         AnalyticsTransmissionTarget childTarget = mChildrenTargets.get(transmissionTargetToken);
         if (childTarget == null) {
-            childTarget = new AnalyticsTransmissionTarget(transmissionTargetToken);
+            childTarget = new AnalyticsTransmissionTarget(transmissionTargetToken, this);
             mChildrenTargets.put(transmissionTargetToken, childTarget);
         }
         return childTarget;
     }
 
+    public AppCenterFuture<Boolean> isEnabledAsync() {
+        final DefaultAppCenterFuture<Boolean> future = new DefaultAppCenterFuture<>();
+        Analytics.getInstance().postCommand(new Runnable() {
+
+            @Override
+            public void run() {
+                future.complete(isEnabled());
+            }
+        }, future, false);
+        return future;
+    }
+
+    public AppCenterFuture<Void> setEnabledAsync(final boolean enabled) {
+        final DefaultAppCenterFuture<Void> future = new DefaultAppCenterFuture<>();
+        Analytics.getInstance().postCommand(new Runnable() {
+
+            @Override
+            public void run() {
+
+                /*
+                 * Like the relation between AppCenter and Analytics, we cannot change state if one of the parent is disabled.
+                 * If this callback is called then it was already checked that AppCenter and Analytics are both enabled.
+                 */
+                if (areAncestorsEnabled()) {
+
+
+                    /* Propagate state to this instance then all descendants without a recursive call. */
```

Comment:
Any reasons why we are not doing cascading calls to child.setEnabled(enabled)?

Reply:
Or is it just optimisation ?

Reply:
We are

Reply:
Ah maybe because child will call isEnabled on their parent.

Reply:
Yes not using recursive calls is a habit.